src/pkg/backup/details/details.go

```diff
@@ -13,9 +13,84 @@ import (
 
 	"github.com/alcionai/corso/src/cli/print"
 	"github.com/alcionai/corso/src/internal/common"
+	"github.com/alcionai/corso/src/internal/version"
 	"github.com/alcionai/corso/src/pkg/path"
 )
 
+// LocationIDer provides access to location information but guarantees that it
+// can also generate a unique location (among items in the same service but
+// possibly across data types within the service) that can be used as a key in
+// maps and other structures. The unique location may be different than
+// InDetails, the location used in backup details.
+type LocationIDer interface {
+	ID() *path.Builder
+	InDetails() *path.Builder
+}
+
+type uniqueLoc struct {
+	pb          *path.Builder
+	prefixElems int
+}
+
+func (ul uniqueLoc) ID() *path.Builder {
+	return ul.pb
+}
+
+func (ul uniqueLoc) InDetails() *path.Builder {
+	return path.Builder{}.Append(ul.pb.Elements()[ul.prefixElems:]...)
+}
+
+// Having service-specific constructors can be kind of clunky, but in this case
+// I think they'd be useful to ensure the proper args are used since this
+// path.Builder is used as a key in some maps.
+
+// NewExchangeLocationIDer builds a LocationIDer for the given category and
+// folder path. The path denoted by the folders should be unique within the
+// category.
+func NewExchangeLocationIDer(
+	category path.CategoryType,
+	escapedFolders ...string,
+) LocationIDer {
```

Comment:
Caught this on second pass.  I'm a big fan of the `accept interfaces, return structs` idiom, unless it's otherwise necessary.  I don't see any necessity here to return the core interface.  Would be more idiomatic to return the struct itself.

Reply:
Imo, good to keep things in line with standards.  Plus, having the struct not be exported will both prevent ad-hoc struct generation and ensure any func which consumes it will prefer an interface over the struct itself.  Overall a safe move that matches golang expectations.